src/app/api/products/favorite-products/[productId]/route.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  response200,
+  response400,
+  response404,
+  response409,
+  response500,
+} from "@/app/utils";
+import { addFavorite, removeFavorite } from "@/model/FavoriteProductsModel";
+import { getProduct } from "@/model/ProductModel";
+import getUserServerSide from "@/services/server/UserService";
+
+export async function PUT(
+  request: Request,
+  {
+    params,
+  }: {
+    params: { productId: string };
+  }
+) {
+  let user = await getUserServerSide();
+
+  if (user === undefined) {
+    return response400("User not logged in ");
+  }
+
+  try {
+    let product = await getProduct(params.productId);
+    if (product === undefined) {
+      return response404("There is no product with this ID");
+    }
+
+    await addFavorite(user.id, params.productId);
+    return response200({
+      message: "Product added to favorites",
```

Comment:
Let's respond with all the product information. I know we don't need it in the UI, but maybe we will in the future.